framework/src/meshgenerators/ParsedCurveGenerator.C

```diff
@@ -228,6 +236,20 @@ ParsedCurveGenerator::generate()
     elem->subdomain_id() = 1;
   }
 
+  if (_node_set_boundaries.size())
+  {
+    // Add boundary nodesets to boundary info
+    BoundaryInfo & boundary_info = mesh->get_boundary_info();
+    int i = 0;
+    for (auto & side_name : _node_set_boundaries)
+      boundary_info.nodeset_name(i++) = side_name;
+
+    boundary_info.add_node(*nodes.begin(), boundary_info.get_id_by_name(_node_set_boundaries[0]));
+    if (_node_set_boundaries.size() > 1)
+      boundary_info.add_node(*(nodes.end() - 1),
+                             boundary_info.get_id_by_name(_node_set_boundaries[1]));
```

Comment:
I would just save the two boundary IDs in a local variable and avoid the `get_id_by_name` lookups

Reply:
it s a cheap lookup there's only 2 ids